libs/MeshKernel/include/MeshKernel/BilinearInterpolationOnGriddedSamples.hpp

```diff
@@ -81,21 +82,164 @@ namespace meshkernel
     };
+
+    template <typename values_type>
+    BilinearInterpolationOnGriddedSamples<values_type>::BilinearInterpolationOnGriddedSamples(const Mesh2D& mesh,
+                                                                                              UInt numXCoord,
+                                                                                              UInt numYCoord,
+                                                                                              const Point& origin,
+                                                                                              double cellSize,
+                                                                                              const void* values) : m_mesh(mesh),
+                                                                                                                    m_numXCoord(numXCoord),
+                                                                                                                    m_numYCoord(numYCoord),
+                                                                                                                    m_origin(origin),
+                                                                                                                    m_cellSize(cellSize),
+                                                                                                                    m_isCellSizeConstant(true)
+    {
+        m_values = static_cast<const values_type*>(values);
+    }
+
+    template <typename values_type>
+    BilinearInterpolationOnGriddedSamples<values_type>::BilinearInterpolationOnGriddedSamples(const Mesh2D& mesh,
+                                                                                              const std::vector<double>& xCoordinates,
+                                                                                              const std::vector<double>& yCoordinates,
+                                                                                              const void* values) : m_mesh(mesh),
+                                                                                                                    m_numXCoord(static_cast<UInt>(xCoordinates.size())),
+                                                                                                                    m_numYCoord(static_cast<UInt>(yCoordinates.size())),
+                                                                                                                    m_xCoordinates(xCoordinates),
+                                                                                                                    m_yCoordinates(yCoordinates),
+                                                                                                                    m_isCellSizeConstant(false)
+    {
+        m_values = static_cast<const values_type*>(values);
+    }
+
+    template <typename values_type>
```

Comment:
General question about this method: if the number of nodes does not change, are the 3 for lops necessary?